app/models/group.rb

```diff
@@ -233,6 +233,10 @@ def archivable?
     # !archived? && children_without_deleted.none?
   end
 
+  def self_registration_active?
+    self_registration_role_type.present? && Settings.groups&.self_registration&.enabled
+  end
+
```

Comment:
validate value of self_registration_role_type (exclusion in Role::Types::WritingPermissions)